From 348c047afd7b3735550b9640b6230e9555435427 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Mon, 22 Jan 2024 21:58:06 +0200
Subject: [PATCH] ACS: More consistent checking of the best channel pointer

It looks like best might be NULL in some cases, so check for this
explicitly before trying to dereference it for a debug print.

Fixes: 733de85680a4 ("ACS: Fix not selecting the best channel in the segment")
Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 src/ap/acs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/acs.c b/src/ap/acs.c
index 7da95d6c2..28b0ba71c 100644
--- a/src/ap/acs.c
+++ b/src/ap/acs.c
@@ -963,7 +963,7 @@ acs_find_ideal_chan_mode(struct hostapd_iface *iface,
 		 * crowded primary channel if one was found in the segment */
 		if (iface->current_mode &&
 		    iface->current_mode->mode == HOSTAPD_MODE_IEEE80211A &&
-		    chan != best) {
+		    best && chan != best) {
 			wpa_printf(MSG_DEBUG,
 				   "ACS: promoting channel %d over %d (less interference %Lg/%Lg)",
 				   best->chan, chan->chan,
-- 
2.43.0

